vllm/worker/habana_model_runner.py

```diff
@@ -77,6 +87,78 @@ def find_bucket(value: int, config: Tuple[int, int, int]):
     return result
 
 
+def subtuple(obj: object, typename: str, to_copy: List[str], to_override: Dict[str, object] = {}):
+    if obj is None:
+        return None
+    fields = set(to_copy) | set(to_override.keys())
+    values = {f: to_override.get(f, getattr(obj, f)) for f in fields}
+    if typename not in _TYPE_CACHE:
+        _TYPE_CACHE[typename] = collections.namedtuple(typename, ' '.join(fields))
+    return _TYPE_CACHE[typename](**values)
+
+
+def align_workers(value, op):
+    group = get_cpu_world_group()
+    world_size = torch.distributed.get_world_size()
+    if world_size <= 1:
+        return value
+    value_t = torch.tensor(value, device='cpu')
+    torch.distributed.all_reduce(value_t, op=op, group=group)
+    return value_t.item()
+
+
+class HpuModelAdapter():
+    def __init__(self, model):
+        self.model = model
+
+    def _set_attn_bias(self, attn_metadata, batch_size, seq_len, device, dtype):
+        prefill_metadata = attn_metadata.prefill_metadata
+        if prefill_metadata is None:
+            return attn_metadata
+        #FIXME: Restore alibi support
+        #if self.alibi_slopes is None:
+        if True:
```

Comment:
nit: can we throw an exception/make an assertion instead of if True?

Reply:
it's a bit of extra work as currently ModelAdapter has no access to alibi_slopes so we'd need to figure out first how to access that. At least in my head, fixing alibi is something that would probably need a better design anyway, so I'm not sure if it's worth fixing at this point.